src/containers/NFT/NFTHeader/Details.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { CopyToClipboard } from '../../shared/components/CopyToClipboard'
+import './styles.scss'
+import { formatPrice } from '../../shared/utils'
+
+interface Props {
+  data: {
+    NFTId: string
+    ledgerIndex: number | undefined
+    owner: string
+    isBurned: boolean | undefined
+    flags: string[] | undefined
+    transferFee: number
+    issuer: string
+    NFTTaxon: number
+    NFTSequence: number | undefined
+    uri: string | undefined
+    validated: boolean | undefined
+    status: string | undefined
+    warnings: string[] | undefined
+    minted: string | undefined
+    domain: string | undefined
+    emailHash: string | undefined
+  }
+}
+
+const Details = ({ data }: Props) => {
+  const { minted, domain, emailHash, NFTTaxon, uri, transferFee } = data
+  const { t } = useTranslation()
+
+  const formattedFee =
+    transferFee === 0
+      ? `${transferFee}%`
+      : `${formatPrice(transferFee / 1000, {
+          currency: 'none',
+          decimals: 5,
+          padding: 3,
+        })}%`
+
+  const abbrvEmail =
+    emailHash &&
+    (emailHash.length > 20 ? emailHash.slice(0, 20).concat('...') : emailHash)
```

Comment:
this is too complex for an expression in my opinion, since you have a ternary nested within a boolean. it should be its own function. furthermore, what is the expected behavior if `emailHash` is falsey? if it is `false` or `null` then `abbrvEmail` will be `false` or `null`. that means the type of this variable could be:

`undefined | null | false | string`

that's not right.

Reply:
this should be done in css anyway basing it on character counts means it is tied for font size and other styles.